duckduckgo/macos-browser
# Navigation failure handled in navigation bar buttons and their menus

Task/Issue URL: https://app.asana.com/0/1177771139624306/1201217438689718/f

**Description**:
Making back/forward navigation buttons and their menus able to cope with navigation error.

**Steps to test this PR**:
Test back/forward buttons and their menus in regular usage
1. Visit a few urls
2. Test back and forward buttons
2. Display back or forward button menus
3. Navigate to a website using the menu

Test closing back to the parent tab using the back button
1. Open a website with CMD + SHIFT + click on a link
2. Make sure the back button menu contains current website and parent website
3. Close the tab by clicking on the back button
4. Repeat the test and close the tab by selecting "Close and return to ..." menu item

Test going back after navigation error
1. Visit https://badssl.com/
2. Click on "Expired"
3. Make sure back button menu contains error page as the current item + badssl.com as a previous item 
4. Go back to badssl.com: 1) by clicking on back, 2) by selecting the menu item

**Testing checklist**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
